JEI + Enchanter causes crash

### Issue description

Hi Devs!

I had a player report that they crashed whenever attempting to use the Oritech Enchanter.

I tested the Enchanter with oritech only, and then added JEI - and the crash was consistent (due to 'The game crashed: rendering screen
Error: java.lang.IllegalArgumentException: height must be >= 0**')

We are on Neoforge 1.21.1 on the newest .145 update

### Steps to reproduce

_No response_

### Minecraft version

1.21.1 (Latest)

### Platform

Fabric

### Loader version

21.1.143

### Fabric API version

0.107.0+2.0.26

### Oritech version

0.45.5

### Other relevant versions

Adding in JEI and then opening the enchanter.

### If a (crash)log is relevant for this issue, link it here: (It's almost always relevant)

_No response_

I also have this issue, my crash log look like this.

```
[00:53:33] [Render thread/INFO]: [FANCYMENU] ScreenCustomizationLayer registered: rearth.oritech.client.ui.EnchanterScreen
[00:53:33] [Render thread/ERROR]: Reported exception thrown!
net.minecraft.util.crash.CrashException: Rendering screen
	at knot//MC/net.minecraft.client.render.GameRenderer.render(GameRenderer.java:922) ~[client-intermediary.jar:?]
	at knot//MC/net.minecraft.client.MinecraftClient.render(MinecraftClient.java:1285) ~[client-intermediary.jar:?]
	at knot//MC/net.minecraft.client.MinecraftClient.run(MinecraftClient.java:882) [client-intermediary.jar:?]
	at knot/net.minecraft.client.main.Main.main(Main.java:256) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480) [fabric-loader-0.16.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.16.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.16.10.jar:?]
Caused by: java.lang.IllegalArgumentException: height must be >= 0
	at knot/com.google.common.base.Preconditions.checkArgument(Preconditions.java:143) ~[guava-32.1.2-jre.jar:?]
	at knot/mezz.jei.common.util.ImmutableRect2i.<init>(ImmutableRect2i.java:26) ~[jei-1.21.1-fabric-19.21.0.247.jar:?]
	at knot/mezz.jei.common.util.ImmutableRect2i.<init>(ImmutableRect2i.java:20) ~[jei-1.21.1-fabric-19.21.0.247.jar:?]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source) ~[?:?]
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:?]
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining(Unknown Source) ~[?:?]
	at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Unknown Source) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline.collect(Unknown Source) ~[?:?]
	at knot/mezz.jei.gui.events.GuiEventHandler.onDrawScreenPost(GuiEventHandler.java:84) ~[jei-1.21.1-fabric-19.21.0.247.jar:?]
	at knot/mezz.jei.fabric.startup.EventRegistration.afterRender(EventRegistration.java:92) ~[jei-1.21.1-fabric-19.21.0.247.jar:?]
	at knot/net.fabricmc.fabric.impl.client.screen.ScreenEventFactory.lambda$createAfterRenderEvent$4(ScreenEventFactory.java:48) ~[fabric-screen-api-v1-2.0.25+8b68f1c719-bdc9794a53021965.jar:?]
	at knot//MC/net.minecraft.client.render.GameRenderer.wrapOperation$fea000$fabric-screen-api-v1$onRenderScreen(GameRenderer.java:6587) ~[client-intermediary.jar:?]
	at knot//MC/net.minecraft.client.render.GameRenderer.mixinextras$bridge$wrapOperation$fea000$fabric-screen-api-v1$onRenderScreen$290(GameRenderer.java) ~[client-intermediary.jar:?]
	at knot//MC/net.minecraft.client.render.GameRenderer.wrapOperation$fho000$fancymenu$wrapRenderScreenFancyMenu(GameRenderer.java:8072) ~[client-intermediary.jar:?]
	at knot//MC/net.minecraft.client.render.GameRenderer.mixinextras$bridge$wrapOperation$fho000$fancymenu$wrapRenderScreenFancyMenu$293(GameRenderer.java) ~[client-intermediary.jar:?]
	at knot//MC/net.minecraft.client.render.GameRenderer.wrapOperation$goj000$konkrete$wrapRenderScreen_Konkrete(GameRenderer.java:14123) ~[client-intermediary.jar:?]
	at knot//MC/net.minecraft.client.render.GameRenderer.render(GameRenderer.java:913) ~[client-intermediary.jar:?]
	... 6 more
```

--- a/common/src/main/java/rearth/oritech/client/ui/AcceleratorScreen.java
+++ b/common/src/main/java/rearth/oritech/client/ui/AcceleratorScreen.java
@@ -91,7 +91,7 @@ public class AcceleratorScreen extends BasicMachineScreen<AcceleratorScreenHandl
                 text = text.formatted(Formatting.BOLD, Formatting.BLACK);
                 titleLabel.horizontalTextAlignment(HorizontalAlignment.LEFT).text(text);
                 speedLabel.text(Text.translatable("text.oritech.accelerator.ui.waiting").formatted(Formatting.BLACK));
-                statusLabel.text(Text.literal(""));
+                statusLabel.text(Text.literal(" "));
             }
             case ERROR -> {
                 titleLabel.text(Text.translatable("text.oritech.accelerator.ui.error").formatted(Formatting.BLACK, Formatting.BOLD));

--- a/common/src/main/java/rearth/oritech/client/ui/EnchanterScreen.java
+++ b/common/src/main/java/rearth/oritech/client/ui/EnchanterScreen.java
@@ -87,7 +87,7 @@ public class EnchanterScreen extends BasicMachineScreen<EnchanterScreenHandler> 
         
         var statistics = handler.enchanter.statistics;
         if (statistics.equals(EnchanterBlockEntity.EnchanterStatistics.EMPTY)) {
-            statisticsLabel.text(Text.literal(""));
+            statisticsLabel.text(Text.literal(" "));
         } else {
             statisticsLabel.text(Text.translatable("title.oritech.enchanter.catalysts", statistics.availableCatalysts(), statistics.requiredCatalysts()).formatted(Formatting.DARK_GRAY));
         }

--- a/common/src/main/java/rearth/oritech/client/ui/InventoryProxyScreen.java
+++ b/common/src/main/java/rearth/oritech/client/ui/InventoryProxyScreen.java
@@ -44,7 +44,7 @@ public class InventoryProxyScreen extends BaseOwoHandledScreen<FlowLayout, Inven
         rootComponent.child(spacer);
         
         for (var slot : Objects.requireNonNull(handler.controllerScreen).getGuiSlots()) {
-            var button = Components.button(Text.literal(""), elem -> {
+            var button = Components.button(Text.literal(" "), elem -> {
                 setActiveSlot(slot.index());
             });
             button.renderer(ORITECH_BUTTON);

--- a/common/src/main/java/rearth/oritech/client/ui/PlayerModifierScreen.java
+++ b/common/src/main/java/rearth/oritech/client/ui/PlayerModifierScreen.java
@@ -323,7 +323,7 @@ public class PlayerModifierScreen extends BaseOwoHandledScreen<FlowLayout, Playe
             var title = Components.label(researchState.type.getName().formatted(Formatting.BOLD));
             title.horizontalSizing(Sizing.fill());
             title.horizontalTextAlignment(HorizontalAlignment.CENTER);
-            var status = Components.label(Text.literal(""));
+            var status = Components.label(Text.literal(" "));
             
             panel.child(title);
             panel.child(status.margins(Insets.of(4, 2, 0, 0)));

--- a/common/src/main/java/rearth/oritech/client/ui/RedstoneAddonScreen.java
+++ b/common/src/main/java/rearth/oritech/client/ui/RedstoneAddonScreen.java
@@ -115,7 +115,7 @@ public class RedstoneAddonScreen extends BaseOwoHandledScreen<FlowLayout, Redsto
         buttonContainer.child(title.positioning(Positioning.relative(50, 5)));
         
         for (var slot : slots) {
-            var button = Components.button(Text.literal(""), elem -> {
+            var button = Components.button(Text.literal(" "), elem -> {
                 setActiveSlot(slot.index());
             });
             buttons.add(button);

